core/src/main/java/org/mule/processor/chain/ProcessorExecutorFactory.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.processor.chain;
+
+import org.mule.api.MuleEvent;
+import org.mule.api.processor.MessageProcessor;
+import org.mule.api.processor.ProcessorExecutor;
+import org.mule.api.service.Service;
+import org.mule.api.transport.CompletionHandlerReplyToHandlerAdaptor;
+import org.mule.execution.MessageProcessorExecutionTemplate;
+import org.mule.processor.BlockingProcessorExecutor;
+import org.mule.processor.NonBlockingProcessorExecutor;
+import org.mule.service.processor.ServiceProcessorExecutor;
+
+import java.util.List;
+
+/**
+ * Create correct instance of {@link org.mule.processor.BlockingProcessorExecutor} based on the current {@link org.mule
+ * .api.MuleEvent}
+ * and {@link org.mule.api.construct.FlowConstruct}.
+ */
+public class ProcessorExecutorFactory
```

Comment:
Can do but didn't do this as there will only be a single default implementation and it makes use more verbose. I was following the YAGNI principal ;-)  Maybe name should be changed to Default ProcessorExecutorFactory in case we introduce factory interface in the future.